Libraries/LibWeb/Fetch/Infrastructure/HTTP/Requests.cpp

```diff
@@ -346,7 +346,7 @@ void Request::add_origin_header()
             case ReferrerPolicy::ReferrerPolicy::StrictOriginWhenCrossOrigin:
                 // If request’s origin is a tuple origin, its scheme is "https", and request’s current URL’s scheme is
                 // not "https", then set serializedOrigin to `null`.
-                if (m_origin.has<URL::Origin>() && m_origin.get<URL::Origin>().scheme() == "https"sv && current_url().scheme() != "https"sv)
+                if (m_origin.has<URL::Origin>() && !m_origin.get<URL::Origin>().is_opaque() && m_origin.get<URL::Origin>().scheme() == "https"sv && current_url().scheme() != "https"sv)
```

Comment:
Maybe we should extract the concept [tuple origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-tuple) into a helper like `is_opaque()`? Also I am a bit confused by us checking for only `https` and `http` schemes. The tuple origin just says it should be an ASCII string and the definition I find for that is: _An ASCII string is a string whose code points are all ASCII code points_.

Reply:
The scheme here is a string and we're only checking for HTTPS because this part of the spec is only interested in whether or not the origin's scheme is "https". The check for whether or not this is a tuple origin is only the `!m_origin.get<URL::Origin>().is_opaque()` and nothing else in this line.

Reply:
Yeah, I do not know why I didn't see that when commenting, I got to focused on the tuple origin concept. Disregard my comment